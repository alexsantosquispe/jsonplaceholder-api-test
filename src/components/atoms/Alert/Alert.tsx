import { twMerge } from 'tailwind-merge';

import {
  CheckCircleIcon,
  CloseIcon,
  ExclamationIcon,
  InfoIcon
} from '../../../icons';

interface AlertProps {
  type?: 'success' | 'error' | 'info';
  title: string;
  message: string;
}

const ICONS_MAP = {
  success: <CheckCircleIcon className="text-green-600 dark:text-emerald-500" />,
  error: <ExclamationIcon className="text-red-600 dark:text-red-500" />,
  info: <InfoIcon className="text-primary/90 dark:text-white/90" />
};
const TEXT_STYLES_MAP = {
  success: 'text-emerald-700 dark:text-emerald-500',
  error: 'text-red-700 dark:text-red-500',
  info: 'text-primary/90 dark:text-white/90'
};

export const Alert = ({ type = 'info', title, message }: AlertProps) => {
  const Icon = ICONS_MAP[type];

  return (
    <div
      className={twMerge(
        'flex w-fit items-center justify-between gap-x-3 rounded-xl border p-3 shadow-lg dark:shadow-neutral-800',
        'border-neutral-300 bg-neutral-100 dark:border-white/10 dark:bg-neutral-800'
      )}
    >
      {Icon}
      <div className="flex flex-col gap-2 text-sm">
        <span className={twMerge('font-medium', TEXT_STYLES_MAP[type])}>
          {title}
        </span>
        <span className="text-neutral-500 dark:text-white/80">{message}</span>
      </div>

      <button className="text-primary/60 hover:text-primary self-start hover:cursor-pointer dark:text-white/70 dark:hover:text-white">
        <CloseIcon className="size-5" />
      </button>
    </div>
  );
};
