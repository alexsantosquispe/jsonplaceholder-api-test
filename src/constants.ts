import { Option } from './services/api.types';

export const THEME_KEY = 'theme';

export const ALERT_TIMEOUT_DEFAULT = 3000;

export const ALERT_TYPES = {
  SUCCESS: 'success',
  ERROR: 'error',
  INFO: 'info'
};

export const DIGITS_LENGTH = 3;

export const ORDER_ASC = 'asc';

export const SORT_BY_NAME = 'name';

export const SORT_BY_TASK_TITLE = 'title';

export const INITIAL_PAGE_NUMBER = 1;

export const LAST_PAGE_NUMBER = 10;

export const PAGE_SIZES: Option[] = [
  { value: 10, label: '10' },
  { value: 20, label: '20' },
  { value: 30, label: '30' },
  { value: 50, label: '50' }
];

export const DEFAULT_PAGE_SIZE: Option = PAGE_SIZES[0];

export const COLUMNS_MAP = [
  { key: 'id', label: 'Code', isSortEnabled: true },
  { key: 'user', label: 'User', isSortEnabled: false },
  { key: 'title', label: 'Title', isSortEnabled: true },
  { key: 'completed', label: 'Status', isSortEnabled: true }
];
