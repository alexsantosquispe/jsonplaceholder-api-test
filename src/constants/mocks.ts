import { Option } from '../services/api.types';

export const optionsMock: Option[] = [
  { value: '1', label: 'option 1' },
  { value: '2', label: 'option 2' },
  { value: '3', label: 'option 3' },
  { value: '4', label: 'option 4' },
  { value: '5', label: 'option 5' },
  { value: '6', label: 'option 6' }
];
