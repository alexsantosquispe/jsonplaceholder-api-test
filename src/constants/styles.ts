export const INPUT_FIELD_STYLES =
  'w-full rounded-lg border border-gray-200 px-4 py-2 text-sm dark:border-white/30 focus:ring-primary/60 focus:border-primary/60 focus:ring-1 focus:outline-none dark:focus:border-white/65 dark:focus:ring-white/65';

export const INPUT_FIELD_ERROR = 'text-xs text-rose-600 dark:text-rose-400';

export const STRIPPED_BG =
  'md:bg-[repeating-linear-gradient(135deg,_var(--pattern-fg)_0,_var(--pattern-fg)_1px,_transparent_0,_transparent_50%)] md:bg-[size:10px_10px] md:bg-fixed md:dark:bg-[repeating-linear-gradient(135deg,_var(--pattern-fg-dark)_0,_var(--pattern-fg-dark)_1px,_transparent_0,_transparent_50%)]';
