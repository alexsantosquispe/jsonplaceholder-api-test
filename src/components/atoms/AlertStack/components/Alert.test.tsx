import { render } from '@testing-library/react';

import { Alert } from './Alert';
import { ALERT_TYPES } from '../../../../constants';

describe('Alert', () => {
  const onCloseMock = jest.fn();
  const props = {
    type: ALERT_TYPES.INFO,
    title: 'Info',
    message: 'Info alert message',
    onClose: onCloseMock
  };

  it('should render the component correctly alert info', () => {
    const alert = render(<Alert {...props} />);

    expect(alert).toMatchSnapshot();
  });

  it('should render the component correctly alert success', () => {
    const alert = render(<Alert {...props} type={ALERT_TYPES.SUCCESS} />);

    expect(alert).toMatchSnapshot();
  });

  it('should render the component correctly alert error', () => {
    const alert = render(<Alert {...props} type={ALERT_TYPES.ERROR} />);

    expect(alert).toMatchSnapshot();
  });
});
