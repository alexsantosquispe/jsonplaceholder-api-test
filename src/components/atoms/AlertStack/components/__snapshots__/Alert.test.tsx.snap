// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render the component correctly alert error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="shadow-primary/30 flex min-w-[18.75rem] items-center justify-between gap-x-2 rounded-lg border px-3 py-2 shadow-lg dark:shadow-white/15 border-neutral-200 bg-neutral-50 dark:border-white/10 dark:bg-neutral-800"
        data-testid="alert"
      >
        <svg
          class="size-6 text-red-600 dark:text-red-500"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12ZM12 8.25a.75.75 0 0 1 .75.75v3.75a.75.75 0 0 1-1.5 0V9a.75.75 0 0 1 .75-.75Zm0 8.25a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="flex w-full flex-col text-sm"
        >
          <span
            class="text-left font-medium text-red-700 dark:text-red-500"
          >
            Info
          </span>
          <span
            class="text-neutral-600 dark:text-white/80"
          >
            Info alert message
          </span>
        </div>
        <button
          class="text-primary/60 hover:text-primary self-start hover:cursor-pointer dark:text-white/70 dark:hover:text-white"
        >
          <svg
            class="size-5"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18 18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="shadow-primary/30 flex min-w-[18.75rem] items-center justify-between gap-x-2 rounded-lg border px-3 py-2 shadow-lg dark:shadow-white/15 border-neutral-200 bg-neutral-50 dark:border-white/10 dark:bg-neutral-800"
      data-testid="alert"
    >
      <svg
        class="size-6 text-red-600 dark:text-red-500"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12ZM12 8.25a.75.75 0 0 1 .75.75v3.75a.75.75 0 0 1-1.5 0V9a.75.75 0 0 1 .75-.75Zm0 8.25a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="flex w-full flex-col text-sm"
      >
        <span
          class="text-left font-medium text-red-700 dark:text-red-500"
        >
          Info
        </span>
        <span
          class="text-neutral-600 dark:text-white/80"
        >
          Info alert message
        </span>
      </div>
      <button
        class="text-primary/60 hover:text-primary self-start hover:cursor-pointer dark:text-white/70 dark:hover:text-white"
      >
        <svg
          class="size-5"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18 18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Alert should render the component correctly alert info 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="shadow-primary/30 flex min-w-[18.75rem] items-center justify-between gap-x-2 rounded-lg border px-3 py-2 shadow-lg dark:shadow-white/15 border-neutral-200 bg-neutral-50 dark:border-white/10 dark:bg-neutral-800"
        data-testid="alert"
      >
        <svg
          class="size-6 text-primary/90 dark:text-white/90"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12Zm8.706-1.442c1.146-.573 2.437.463 2.126 1.706l-.709 2.836.042-.02a.75.75 0 0 1 .67 1.34l-.04.022c-1.147.573-2.438-.463-2.127-1.706l.71-2.836-.042.02a.75.75 0 1 1-.671-1.34l.041-.022ZM12 9a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="flex w-full flex-col text-sm"
        >
          <span
            class="text-left font-medium text-primary/90 dark:text-white/90"
          >
            Info
          </span>
          <span
            class="text-neutral-600 dark:text-white/80"
          >
            Info alert message
          </span>
        </div>
        <button
          class="text-primary/60 hover:text-primary self-start hover:cursor-pointer dark:text-white/70 dark:hover:text-white"
        >
          <svg
            class="size-5"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18 18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="shadow-primary/30 flex min-w-[18.75rem] items-center justify-between gap-x-2 rounded-lg border px-3 py-2 shadow-lg dark:shadow-white/15 border-neutral-200 bg-neutral-50 dark:border-white/10 dark:bg-neutral-800"
      data-testid="alert"
    >
      <svg
        class="size-6 text-primary/90 dark:text-white/90"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12Zm8.706-1.442c1.146-.573 2.437.463 2.126 1.706l-.709 2.836.042-.02a.75.75 0 0 1 .67 1.34l-.04.022c-1.147.573-2.438-.463-2.127-1.706l.71-2.836-.042.02a.75.75 0 1 1-.671-1.34l.041-.022ZM12 9a.75.75 0 1 0 0-1.5.75.75 0 0 0 0 1.5Z"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="flex w-full flex-col text-sm"
      >
        <span
          class="text-left font-medium text-primary/90 dark:text-white/90"
        >
          Info
        </span>
        <span
          class="text-neutral-600 dark:text-white/80"
        >
          Info alert message
        </span>
      </div>
      <button
        class="text-primary/60 hover:text-primary self-start hover:cursor-pointer dark:text-white/70 dark:hover:text-white"
      >
        <svg
          class="size-5"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18 18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Alert should render the component correctly alert success 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="shadow-primary/30 flex min-w-[18.75rem] items-center justify-between gap-x-2 rounded-lg border px-3 py-2 shadow-lg dark:shadow-white/15 border-neutral-200 bg-neutral-50 dark:border-white/10 dark:bg-neutral-800"
        data-testid="alert"
      >
        <svg
          class="size-6 text-green-600 dark:text-emerald-500"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12Zm13.36-1.814a.75.75 0 1 0-1.22-.872l-3.236 4.53L9.53 12.22a.75.75 0 0 0-1.06 1.06l2.25 2.25a.75.75 0 0 0 1.14-.094l3.75-5.25Z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="flex w-full flex-col text-sm"
        >
          <span
            class="text-left font-medium text-emerald-700 dark:text-emerald-500"
          >
            Info
          </span>
          <span
            class="text-neutral-600 dark:text-white/80"
          >
            Info alert message
          </span>
        </div>
        <button
          class="text-primary/60 hover:text-primary self-start hover:cursor-pointer dark:text-white/70 dark:hover:text-white"
        >
          <svg
            class="size-5"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 18 18 6M6 6l12 12"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="shadow-primary/30 flex min-w-[18.75rem] items-center justify-between gap-x-2 rounded-lg border px-3 py-2 shadow-lg dark:shadow-white/15 border-neutral-200 bg-neutral-50 dark:border-white/10 dark:bg-neutral-800"
      data-testid="alert"
    >
      <svg
        class="size-6 text-green-600 dark:text-emerald-500"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12Zm13.36-1.814a.75.75 0 1 0-1.22-.872l-3.236 4.53L9.53 12.22a.75.75 0 0 0-1.06 1.06l2.25 2.25a.75.75 0 0 0 1.14-.094l3.75-5.25Z"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="flex w-full flex-col text-sm"
      >
        <span
          class="text-left font-medium text-emerald-700 dark:text-emerald-500"
        >
          Info
        </span>
        <span
          class="text-neutral-600 dark:text-white/80"
        >
          Info alert message
        </span>
      </div>
      <button
        class="text-primary/60 hover:text-primary self-start hover:cursor-pointer dark:text-white/70 dark:hover:text-white"
      >
        <svg
          class="size-5"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 18 18 6M6 6l12 12"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
